src/stream/src/executor/over_window/state/mod.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeSet;
+
+use risingwave_common::types::{Datum, ScalarImpl};
+use risingwave_expr::expr::WindowFuncKind;
+use smallvec::SmallVec;
+
+use super::call::WindowFuncCall;
+use super::MemcmpEncoded;
+
+mod lag;
+mod lead;
+
+/// Unique and ordered identifier for a row in internal states.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub(super) struct StateKey {
+    pub order_key: ScalarImpl,
```

Comment:
Does this mean that we cannot specify multiple columns here? (sounds like a limitation of the sort executor)

Reply:
Yes, this is a limitation of `SortExecutor`.